### DISCUSSION TITLE
nlbwmon 网络带宽监视器不起作用

### DISCUSSION THREAD

**Discussion by ttttr885** [immortalwrt/immortalwrt#1381] [Q&A] [answered]
请问为什么安装了nlbwmon 却没有数据  用官方的OPenwrt 就没有这一个问题 能麻烦给一下解决方案么

**Comment by KardRi** [comment]
我mt798x的設備也都跑不出nlbwmon，都是無紀錄，不知道是不是hw nat的問題
版本是25.05.2 
nlbwmon 2021-09-01-d82c910c-1 
luci-app-nlbwmon git-24.086.26304-1d8fc03

**Reply by ttttr885** [reply]
https://github.com/brvphoenix/luci-app-wrtbwmon
找了这么多 只有这一个能用  需要关硬件及软件加速

**Reply by KardRi** [reply]
我覺得，或許可以自己compile他的snapshot版?

23.05.x都是2021-09-01-d82c910c-1，snapshot是2024.02.21~c7616bcf-r1

Edit: 好吧，沒用

**Reply by KardRi** [reply]
突然發現是有用的，只是我沒有當下沒有reboot而已

裝的是2024-02-21-c7616bcf-1，23.05.2 kernel 5.15.150編譯出來的nlbwmon

**Reply by sagezhj** [reply]
编译整个固件然后单独提取ipk吗？

**Reply by KardRi** [reply, marked as answer]
先
git clone --depth 1 --branch v23.05.2 https://github.com/immortalwrt/immortalwrt

再手動fork immortalwrt/packages，更新net/nlbwmon的[Makefile](https://github.com/KardRi/packages/commit/d5672e606cb27796bb22da6c44f65501cd1e6851)
然後更改feeds.conf.default的packages部分，make menuconfig勾上nlbwmon，其他隨意

**Reply by ttttr885** [reply]
能不能共享一下你编译的插件 小白一个 不会操作

**Reply by KardRi** [reply]
你是哪種arch? 

之前編譯的aarch64_cortex_a53，適用mt798x / ipq60xx(應該)，版本23.05.2對應內核5.15.150

[nlbwmon_2024-02-21-c7616bcf-1_aarch64_cortex-a53.zip](https://github.com/user-attachments/files/16502901/nlbwmon_2024-02-21-c7616bcf-1_aarch64_cortex-a53.zip)

**Reply by ttttr885** [reply]
谢谢 我N60上能用 但这玩意在统计上差的有点大了

**Reply by KardRi** [reply]
在firewall那邊嘗試關閉software offload或是hardware offload看看

**Reply by ttttr885** [reply]
我知道需要关硬件加速 那这个就失去意义了

**Comment by sagezhj** [comment]
我是x86架构，也有这个问题，想问下各路大神现在是否有解决办法？[wrtbwmon](https://github.com/brvphoenix/luci-app-wrtbwmon)试了不行，安装失败。以下是我安装设备的相关信息：
架构 | Intel(R) N100 x 4C 4T (800.000MHz, 40.0°C)
目标平台 | x86/64
固件版本 | ImmortalWrt 23.05.2 r27625-416c8c5c91 / LuCI openwrt-23.05 branch git-24.079.19999-26165fd
内核版本 | 5.15.150
